### DISCUSSION TITLE
Is `ignore` supposed to disable intellisense features?

### DISCUSSION THREAD

**Discussion by max-muoto** [microsoft/pyright#6351] [General]
My understanding is that the `exclude` array would disable intellisense features for the specified modules, however based on the description of `ignore`, I assumed only errors would be silenced, but intellisense functionality would remain the same?

For example, I'm wildcarding tests as ignored `ignore = ["**/tests"]` in our `pyproject.toml` however this has the side effect of disabling quick import functionality in all of our tests for VSCode users. Mainly wanting to understand if there is a workaround here, or if this is intended.

**VS Code extension or command-line**
CLI tool with the VSCode Pylance extension.

**Comment by erictraut** [comment]
Moving to discussion since this is a question, not a bug report.

**Reply by max-muoto** [reply]
Thanks, apologies there. Wasn't sure if it might be a bug!

**Reply by erictraut** [reply]
The `ignore` feature suppresses diagnostics for that file. 

Some diagnostics have associated code actions. If you suppress the diagnostics, the corresponding code actions won't be available.

**Reply by max-muoto** [reply]
I see, thanks for the info. I know I had a previous inquiry [here](https://github.com/microsoft/pyright/issues/6240), but going back to that would you recommend probably just to shift over to typing comments to opt-in files to solve this? I'm thinking of doing that, but I do like the ability to have our base settings be a bit more custom.

I know you mentioned that here: https://github.com/microsoft/pyright/issues/1443. I suppose one thing I'd want to make sure is that we set up a linting rule to require this on all new files, not sure if that's something you did at Microsoft.
